parley/src/layout/cursor.rs

```diff
@@ -1051,7 +1051,11 @@ enum AnchorBase {
 
 fn cursor_rect<B: Brush>(cluster: &Cluster<'_, B>, at_end: bool, size: f32) -> Rect {
     let line_x = (cluster.visual_offset().unwrap_or_default()
-        + at_end.then(|| cluster.advance()).unwrap_or_default()) as f64;
+        + if at_end {
+            cluster.advance()
+        } else {
+            Default::default()
+        }) as f64;
```

Comment:
Less clear on the code style here; I don't think either the old proposal or the new one is particularly good, although I'm not sure that the new one is better...

I'm really not sure what is best here; I think I'm happy for this change to land as-is, but I'd be keen to try an alternative if you can think of one.

Reply:
I've gone with this.